toolchain/check/handle_statement.cpp

```diff
@@ -27,51 +28,46 @@ auto HandleExpressionStatement(Context& context, Parse::Node /*parse_node*/)
   return true;
 }
 
-auto HandleReturnStatement(Context& context, Parse::Node parse_node) -> bool {
-  CARBON_CHECK(!context.return_scope_stack().empty());
-  auto fn_inst = context.insts().GetAs<SemIR::FunctionDeclaration>(
-      context.return_scope_stack().back());
-  const auto& callable = context.functions().Get(fn_inst.function_id);
-
-  if (context.parse_tree().node_kind(context.node_stack().PeekParseNode()) ==
-      Parse::NodeKind::ReturnStatementStart) {
-    context.node_stack()
-        .PopAndDiscardSoloParseNode<Parse::NodeKind::ReturnStatementStart>();
+auto HandleReturnStatementStart(Context& context, Parse::Node parse_node)
```

Comment:
Would it be worth breaking this out to a separate `handle_return`, maybe leaving behind a small and boring `handle_expression_statement`?

Partly, I assume the BuildReturn functions can only be used here: it feels odd to me that I need to go looking in a separate file for their implementation. Maybe you won't want to re-merge, but even then, it feels odd to me to have `ReturnVarSpecifier` handled here. You'd already said you had trouble finding `return`, so maybe a split would help that?

Reply:
Sure, split `handle_statement.cpp` into `handle_expression_statement.cpp` and `handle_return_statement.cpp`.

I think it's useful to keep a separate, non-`handle_` file for the `return ...` and `returned var` processing, because they share logic. It wouldn't make sense to put `HandlePatternBinding` in the same file as `HandleReturnStatement`, and I don't think we want to have the `handle_*` files expose any functions other than their `Handle*` functions, so I wouldn't want to re-merge.